### PR TITLE
support string data parameter to decode method

### LINKED ISSUE #3: Error: Object: is not a legal argument to this wrapper, cause it doesn't respond to "read"
There don't seem to be a lot of people talking about this codec in the wild, so maybe I'm suffering from a lack of examples.

I have this input:

```
file {
    type => "foo"
    path => "/path1/path2/*"
    codec => "gzip_lines"
}
```

I have installed the plugin, and logstash starts.

I've tested it with a single (valid, text) file in the directory, but I get messages like this:

```
{:timestamp=>"2016-01-06T09:30:16.917000-0800", :message=>"A plugin had an unrecoverable error. Will restart this plugin.\n  Plugin: <LogStash::Inputs::File type=>\"foo\", path=>[\"/path1/path2/*\"], codec=><LogStash::Codecs::GzipLines charset=>\"UTF-8\">, debug=>false, stat_interval=>1, discover_interval=>15, sincedb_write_interval=>15, start_position=>\"end\", delimiter=>\"\\n\">\n  Error: Object: \u001F\u008B\b\u0000ÑA\u008CV\u0000\u0003-Ê1\u000E\u00021\fDÑ\u009ESX¢vAË\u0001¸\u0006 is not a legal argument to this wrapper, cause it doesn't respond to \"read\".", :level=>:error}
```

Googling that error finds me some kindred spirits, but no clear answers.

### PR DESCRIPTION
**Pull request by yaauie**
Resolves #3

### CODE REVIEW

**Comment by jsvd** [comment]
I created #9 to test this, and the  tests pass

File: spec/codecs/gzip_lines_spec.rb [hunk 1]
```diff
@@ -26,18 +26,42 @@ def compress_with_gzip(io)
...
-    it "should create events from a gzip file" do
-      events = []
+    let(:compressed_log) { compress_with_gzip(uncompressed_log) }
+
+    # Legacy tests and implementation _required_ an `IO`-like object to be
+    # provided to `LogStash::Codecs::GzipLines#decode`, despite the Logstash
+    # API specifying that a `String` is provided. Continue supporting this
+    # to avoid breaking implementations that provide an IO.
+    context 'when given a StringIO' do
+      it "should create events from a gzip file" do
```
Comment (jsvd) [inline spec/codecs/gzip_lines_spec.rb]: seems that `from a gzip` file doesn't make much sense here, maybe `retrieve one event per compressed line`

**Review by jsvd — APPROVED** [review]
I added a minor note about accuracy of the assertion description.

overall LGTM, I suggest reviewing and merging #9, rebasing this PR off of it and merging+publishing

**Comment by aaron-kahn** [comment]
I am trying to use this plugin and I am receiving the following error:

> Error: not in gzip format
> Exception: Zlib::GzipFile::Error

The file that I am processing is definitely a gzipped file. And I have confirmed that through the Ruby shell. Here is the config that I am using

```
input {
        file {
                path => "path/*.gz"
                ignore_older => 3600
                mode => "read"
                file_completed_action => "delete"
                codec => "gzip_lines"
        }
}

output {
        stdout {}
}
```

I have even tried downgrading the plugin to v3.0.3 and I indeed see the error that this PR was intended to address:

> Error: Object: This is the first line in the file is not a legal argument to this wrapper, cause it doesn't respond to "read".

I am running Logstash v7.2.0. Have I simply configured something incorrectly?

**Pull request merged**